src/analytics/api/models/events.py

```diff
@@ -79,6 +79,9 @@ def from_bigquery(
             ["site", "datetime", "device"], ascending=True, inplace=True
         )
 
+        dataframe.drop_duplicates(
+            subset=["datetime", "device"], inplace=True, keep="first"
```

Comment:
Awesome @NoahNsimbe , thanks @NoahNsimbe ....
We could also ensure that we address this from the pipeline too, if it is not "costly".
Otherwise, no worries!